The job **`x86_64-gnu-llvm-19`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/14986009530/job/42100009610) [(plain)](https://github.com/rust-lang/rust/commit/44bb04c7af653312f3f29c0eccb09eab240a1c32/checks/42100009610/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
#18 exporting to docker image format
#18 sending tarball 22.0s done
#18 DONE 28.9s
##[endgroup]
Setting extra environment values for docker:  --env ENABLE_GCC_CODEGEN=1 --env GCC_EXEC_PREFIX=/usr/lib/gcc/
[CI_JOB_NAME=x86_64-gnu-llvm-19]
[CI_JOB_NAME=x86_64-gnu-llvm-19]
debug: `DISABLE_CI_RUSTC_IF_INCOMPATIBLE` configured.
---
sccache: Listening on address 127.0.0.1:4226
##[group]Configure the build
configure: processing command line
configure: 
configure: build.configure-args := ['--build=x86_64-unknown-linux-gnu', '--llvm-root=/usr/lib/llvm-19', '--enable-llvm-link-shared', '--set', 'rust.randomize-layout=true', '--set', 'rust.thin-lto-import-instr-limit=10', '--set', 'build.print-step-timings', '--enable-verbose-tests', '--set', 'build.metrics', '--enable-verbose-configure', '--enable-sccache', '--disable-manage-submodules', '--enable-locked-deps', '--enable-cargo-native-static', '--set', 'rust.codegen-units-std=1', '--set', 'dist.compression-profile=balanced', '--dist-compression-formats=xz', '--set', 'rust.lld=false', '--disable-dist-src', '--release-channel=nightly', '--enable-debug-assertions', '--enable-overflow-checks', '--enable-llvm-assertions', '--set', 'rust.verify-llvm-ir', '--set', 'rust.codegen-backends=llvm,cranelift,gcc', '--set', 'llvm.static-libstdcpp', '--set', 'gcc.download-ci-gcc=true', '--enable-new-symbol-mangling']
configure: build.build          := x86_64-unknown-linux-gnu
configure: target.x86_64-unknown-linux-gnu.llvm-config := /usr/lib/llvm-19/bin/llvm-config
configure: llvm.link-shared     := True
configure: rust.randomize-layout := True
configure: rust.thin-lto-import-instr-limit := 10
---
[RUSTC-TIMING] object test:false 2.697
error[E0546]: 'feature' is not an identifier
    --> library/std/src/path.rs:2071:1
     |
2071 | #[stable(feature = "eq-str-for-path", since = "CURRENT_RUSTC_VERSION")]
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0546]: 'feature' is not an identifier
    --> library/std/src/path.rs:2079:1
     |
2079 | #[stable(feature = "eq-str-for-path", since = "CURRENT_RUSTC_VERSION")]
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0546]: 'feature' is not an identifier
    --> library/std/src/path.rs:2087:1
     |
2087 | #[stable(feature = "eq-str-for-path", since = "CURRENT_RUSTC_VERSION")]
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0546]: 'feature' is not an identifier
    --> library/std/src/path.rs:2095:1
     |
2095 | #[stable(feature = "eq-str-for-path", since = "CURRENT_RUSTC_VERSION")]
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0546]: 'feature' is not an identifier
    --> library/std/src/path.rs:3277:1
     |
3277 | #[stable(feature = "eq-str-for-path", since = "CURRENT_RUSTC_VERSION")]
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0546]: 'feature' is not an identifier
    --> library/std/src/path.rs:3286:1
     |
3286 | #[stable(feature = "eq-str-for-path", since = "CURRENT_RUSTC_VERSION")]
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0546]: 'feature' is not an identifier
    --> library/std/src/path.rs:3294:1
     |
3294 | #[stable(feature = "eq-str-for-path", since = "CURRENT_RUSTC_VERSION")]
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0546]: 'feature' is not an identifier
    --> library/std/src/path.rs:3302:1
     |
3302 | #[stable(feature = "eq-str-for-path", since = "CURRENT_RUSTC_VERSION")]
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: implementation has missing stability attribute
    --> library/std/src/path.rs:2072:1
     |
2072 | / impl cmp::PartialEq<str> for PathBuf {
2073 | |     #[inline]
2074 | |     fn eq(&self, other: &str) -> bool {
2075 | |         &*self == other
2076 | |     }
2077 | | }
     | |_^

error: implementation has missing stability attribute
    --> library/std/src/path.rs:2080:1
     |
2080 | / impl cmp::PartialEq<PathBuf> for str {
2081 | |     #[inline]
2082 | |     fn eq(&self, other: &PathBuf) -> bool {
2083 | |         other == self
2084 | |     }
2085 | | }
     | |_^

error: implementation has missing stability attribute
    --> library/std/src/path.rs:2088:1
     |
2088 | / impl cmp::PartialEq<String> for PathBuf {
2089 | |     #[inline]
2090 | |     fn eq(&self, other: &String) -> bool {
2091 | |         **self == **other
2092 | |     }
2093 | | }
     | |_^

error: implementation has missing stability attribute
    --> library/std/src/path.rs:2096:1
     |
2096 | / impl cmp::PartialEq<PathBuf> for String {
2097 | |     #[inline]
2098 | |     fn eq(&self, other: &PathBuf) -> bool {
2099 | |         other == self
2100 | |     }
2101 | | }
     | |_^

error: implementation has missing stability attribute
    --> library/std/src/path.rs:3278:1
     |
3278 | / impl cmp::PartialEq<str> for Path {
3279 | |     #[inline]
3280 | |     fn eq(&self, other: &str) -> bool {
3281 | |         let other: &OsStr = other.as_ref();
...    |
3284 | | }
     | |_^

error: implementation has missing stability attribute
    --> library/std/src/path.rs:3287:1
     |
3287 | / impl cmp::PartialEq<Path> for str {
3288 | |     #[inline]
3289 | |     fn eq(&self, other: &Path) -> bool {
3290 | |         other == self
3291 | |     }
3292 | | }
     | |_^

error: implementation has missing stability attribute
    --> library/std/src/path.rs:3295:1
     |
3295 | / impl cmp::PartialEq<String> for Path {
3296 | |     #[inline]
3297 | |     fn eq(&self, other: &String) -> bool {
3298 | |         self == &*other
3299 | |     }
3300 | | }
     | |_^

error: implementation has missing stability attribute
    --> library/std/src/path.rs:3303:1
     |
3303 | / impl cmp::PartialEq<Path> for String {
3304 | |     #[inline]
3305 | |     fn eq(&self, other: &Path) -> bool {
3306 | |         other == self
3307 | |     }
3308 | | }
     | |_^

For more information about this error, try `rustc --explain E0546`.
```

</details>

--- a/library/std/src/path.rs
+++ b/library/std/src/path.rs
@@ -2068,6 +2068,38 @@ impl PartialEq for PathBuf {
     }
 }
 
+#[stable(feature = "eq_str_for_path", since = "CURRENT_RUSTC_VERSION")]
+impl cmp::PartialEq<str> for PathBuf {
+    #[inline]
+    fn eq(&self, other: &str) -> bool {
+        &*self == other
+    }
+}
+
+#[stable(feature = "eq_str_for_path", since = "CURRENT_RUSTC_VERSION")]
+impl cmp::PartialEq<PathBuf> for str {
+    #[inline]
+    fn eq(&self, other: &PathBuf) -> bool {
+        other == self
+    }
+}
+
+#[stable(feature = "eq_str_for_path", since = "CURRENT_RUSTC_VERSION")]
+impl cmp::PartialEq<String> for PathBuf {
+    #[inline]
+    fn eq(&self, other: &String) -> bool {
+        **self == **other
+    }
+}
+
+#[stable(feature = "eq_str_for_path", since = "CURRENT_RUSTC_VERSION")]
+impl cmp::PartialEq<PathBuf> for String {
+    #[inline]
+    fn eq(&self, other: &PathBuf) -> bool {
+        other == self
+    }
+}
+
 #[stable(feature = "rust1", since = "1.0.0")]
 impl Hash for PathBuf {
     fn hash<H: Hasher>(&self, h: &mut H) {
@@ -3239,6 +3271,39 @@ impl PartialEq for Path {
     #[inline]
     fn eq(&self, other: &Path) -> bool {
         self.components() == other.components()
+    }
+}
+
+#[stable(feature = "eq_str_for_path", since = "CURRENT_RUSTC_VERSION")]
+impl cmp::PartialEq<str> for Path {
+    #[inline]
+    fn eq(&self, other: &str) -> bool {
+        let other: &OsStr = other.as_ref();
+        self == other
+    }
+}
+
+#[stable(feature = "eq_str_for_path", since = "CURRENT_RUSTC_VERSION")]
+impl cmp::PartialEq<Path> for str {
+    #[inline]
+    fn eq(&self, other: &Path) -> bool {
+        other == self
+    }
+}
+
+#[stable(feature = "eq_str_for_path", since = "CURRENT_RUSTC_VERSION")]
+impl cmp::PartialEq<String> for Path {
+    #[inline]
+    fn eq(&self, other: &String) -> bool {
+        self == &*other
+    }
+}
+
+#[stable(feature = "eq_str_for_path", since = "CURRENT_RUSTC_VERSION")]
+impl cmp::PartialEq<Path> for String {
+    #[inline]
+    fn eq(&self, other: &Path) -> bool {
+        other == self
     }
 }
 

--- a/tests/ui/inference/issue-72616.stderr
+++ b/tests/ui/inference/issue-72616.stderr
@@ -6,14 +6,10 @@ LL |         if String::from("a") == "a".try_into().unwrap() {}
    |                              |
    |                              type must be known at this point
    |
-   = note: cannot satisfy `String: PartialEq<_>`
-   = help: the following types implement trait `PartialEq<Rhs>`:
-             `String` implements `PartialEq<&str>`
-             `String` implements `PartialEq<ByteStr>`
-             `String` implements `PartialEq<ByteString>`
-             `String` implements `PartialEq<Cow<'_, str>>`
-             `String` implements `PartialEq<str>`
-             `String` implements `PartialEq`
+   = note: multiple `impl`s satisfying `String: PartialEq<_>` found in the following crates: `alloc`, `std`:
+           - impl PartialEq for String;
+           - impl PartialEq<Path> for String;
+           - impl PartialEq<PathBuf> for String;
 help: try using a fully qualified path to specify the expected types
    |
 LL -         if String::from("a") == "a".try_into().unwrap() {}

--- a/tests/ui/suggestions/partialeq_suggest_swap_on_e0277.stderr
+++ b/tests/ui/suggestions/partialeq_suggest_swap_on_e0277.stderr
@@ -10,6 +10,8 @@ LL |     String::from("Girls Band Cry") == T(String::from("Girls Band Cry"));
              `String` implements `PartialEq<ByteStr>`
              `String` implements `PartialEq<ByteString>`
              `String` implements `PartialEq<Cow<'_, str>>`
+             `String` implements `PartialEq<Path>`
+             `String` implements `PartialEq<PathBuf>`
              `String` implements `PartialEq<str>`
              `String` implements `PartialEq`
    = note: `T` implements `PartialEq<String>`

